src/dishka/container.py

```diff
@@ -213,6 +223,14 @@ def close(self, exception: BaseException | None = None) -> None:
         if errors:
             raise ExitError("Cleanup context errors", errors)  # noqa: TRY003
 
+    @staticmethod
+    def _get_dependency_key(key: Any) -> Any:
```

Comment:
1. NEVER use `staticmethod` unless you really want function to be called on class which is wrong here
2. Additional calls are slow. Avoid introducing additional functions if they are going to be called very often
3. staticmethod is so slow.